test/hotspot/jtreg/compiler/jvmci/jdk.vm.ci.hotspot.test/src/jdk/vm/ci/hotspot/test/TestHotSpotJVMCIRuntime.java

```diff
@@ -170,6 +170,7 @@ public void jniEnomemTest() throws Exception {
             ProcessBuilder pb = ProcessTools.createLimitedTestJavaProcessBuilder(
                 "-XX:+UnlockExperimentalVMOptions",
                 "-XX:+EnableJVMCI",
+                "--add-modules=jdk.internal.vm.ci",
```

Comment:
I stared at this for a while to understand why passing this option was required.  It's a bit confusing that explicitly passing `-XX:+EnableJVMCI` has different effects based on the value of UseJVMCINativeLibrary.  I think that if `EnableJVMCI` is passed on the command line then it should add the module even if libgraal is in use.  So something like:
`if ((!UseJVMCINativeLibrary || FLAG_IS_CMDLINE(EnableJVMCI) && ClassLoader::is_module_observable`

Reply:
I was not aware FLAG_IS_CMDLINE can be used for altering the semantics of a flag but there seems to be at least one precedent for it with [UseCompactObjectHeaders](https://github.com/openjdk/jdk/blob/3dd34517000e4ce1a21619922c62c025f98aad44/src/hotspot/share/runtime/arguments.cpp#L3671). This is quite nice as now nothing needs to change for Truffle users in terms of enabling the Truffle optimized runtime (cc @chumer).